api/src/main/java/com/strategyobject/substrateclient/api/Api.java

```diff
@@ -1,7 +1,33 @@
 package com.strategyobject.substrateclient.api;
 
+import com.strategyobject.substrateclient.pallet.GeneratedPalletResolver;
 import com.strategyobject.substrateclient.rpc.Rpc;
+import com.strategyobject.substrateclient.rpc.RpcImpl;
+import com.strategyobject.substrateclient.transport.ProviderInterface;
+import lombok.val;
 
+/**
+ * Provides the ability to query a node and interact with the Polkadot or Substrate chains.
+ * It allows interacting with blockchain in different ways: using RPC's queries directly or
+ * accessing Pallets and its APIs such storages, transactions, etc.
+ */
 public interface Api {
+    static DefaultApi with(ProviderInterface provider) {
+        val rpc = new RpcImpl(provider);
+
+        return DefaultApi.with(rpc, GeneratedPalletResolver.with(rpc));
+    }
+
+    /**
+     * @return the instance that provides proper API for querying RPC's methods.
+     */
     Rpc rpc();
+
+    /**
+     * Resolves the instance of pallet by its definition.
```

Comment:
Resolves the instance of a pallet by its definition.
 OR
 Resolves a pallet instance by its definition.